base/settings/R/createMultisiteMultiSettings.r

```diff
@@ -1,85 +1,100 @@
 #' @param sitegroupId The Bety ID of the sitegroup to draw from
-#' @param nSite The number of sites to randomly select (without replacement) from the siteGroup. Omit
-#' to use all sites in the group.
-#' @param con,params Bety DB connection or parameters. passed directly to \code{\link[PEcAn.DB]{db.query}}
-#' 
-#' @details 
-#' Starts with a template settings object, and fills in the \code{run} block with site info sampled from
-#' the sitegroup. The template could be fully set up except for the site info, or more or less empty if you 
-#' plan to fill in the other settings later. A \code{\link{MultiSettings}} is created from \code{templateSettings},
-#' \code{nSite} sites (or all of them, if \code{nSite} is unset) are selected from Bety, and their info
-#' is dropped into the \code{MultiSettings}.
-#'   
-#' @return A \code{MultiSettings} object with the same settings as \code{templateSettings} but site
-#' information for the selected sites
-#' 
+#' @param nSite The number of sites to randomly select (without replacement)
+#'   from the siteGroup. Omit to use all sites in the group.
+#' @param con,params Bety DB connection or parameters.
+#'   Passed directly to \code{\link[PEcAn.DB]{db.query}}
+#'
+#' @details
+#' Starts with a template settings object, and fills in the \code{run} block
+#'  with site info sampled from the sitegroup.
+#'  The template could be fully set up except for the site info,
+#'  or more or less empty if you plan to fill in the other settings later.
+#'  A \code{\link{MultiSettings}} is created from \code{templateSettings},
+#'  \code{nSite} sites (or all of them, if \code{nSite} is unset) are selected
+#'  from Bety, and their info is dropped into the \code{MultiSettings}.
+#'
+#' @return A \code{MultiSettings} object with the same settings as
+#'  \code{templateSettings} but site information for the selected sites
+#'
 #' @author Ryan Kelly
 #' @export
-#' 
+#'
 #' @example examples/examples.MultiSite.MultiSettings.r
-createSitegroupMultiSettings = function(templateSettings, sitegroupId, nSite, con=NULL, params=templateSettings$database$bety) {
-  query <- paste("SELECT site_id FROM sitegroups_sites WHERE sitegroup_id =", sitegroupId)
-  allSites <- PEcAn.DB::db.query(query, con=con, params=params)   
-  
-  if(missing(nSite))
+createSitegroupMultiSettings <- function(
+    templateSettings,
+    sitegroupId,
+    nSite,
+    con = NULL,
+    params = templateSettings$database$bety) {
+  query <- paste(
+    "SELECT site_id FROM sitegroups_sites WHERE sitegroup_id =", sitegroupId)
+  allSites <- PEcAn.DB::db.query(query, con = con, params = params)
+
+  if (missing(nSite)) {
     siteIds <- allSites$site_id
-  else 
-    siteIds <- sample(allSites$site_id, nSite, replace=FALSE)
+  } else {
+    siteIds <- sample(allSites$site_id, nSite, replace = FALSE)
+  }
 
-  settings <- createMultiSiteSettings(templateSettings, siteIds)
+  createMultiSiteSettings(templateSettings, siteIds)
```

Comment:
this creates the actual settings variable and overwrites the global version I think. I'm not sure this does not break things if we remove it.

Reply:
I'm *fairly* sure it doesn't break anything, but it's good to check. Does this reasoning seem sound?
* `<-` binds in the current environment; we'd need to use `<<-` to affect the global object.
* All calls to `createSitegroupMultiSettings` in the codebase assign their result to an opbject named `settings`, so even if we were globally assigning it here we'd overwrite it on return.
* OK, technically this does change behavior: Since `<-(x, value)` returns `value` invisibly and we were using that assignment line as the implicit return value of this function, this change mean createSitegroupMultiSettings now returns its result visibly rather than invisibly :)